File: web/types/chat/ChatStreamErrorEvent.ts
// enums
import { ChatStreamEventTypeEnum } from '@/enums/chat';

// types
import type BaseChatStreamEvent from './BaseChatStreamEvent';
import type { BaseErrorResponse } from '@/types/errors';

interface ChatStreamErrorEvent extends BaseChatStreamEvent {
  error: BaseErrorResponse;
  type: ChatStreamEventTypeEnum.Error;
}

export default ChatStreamErrorEvent;

File: api/schemas/chat/chat_stream_error_event.py
from typing import Literal
from uuid import UUID

from pydantic import BaseModel

from api import enums
from api.schemas.errors import BaseErrorResponse


class ChatStreamErrorEvent(BaseModel):
    error: BaseErrorResponse
    session_id: UUID
    type: Literal[enums.chat.ChatStreamEventTypeEnum.ERROR] = enums.chat.ChatStreamEventTypeEnum.ERROR
